src/components/drawing/index.js

```diff
@@ -298,11 +374,12 @@ drawing.pointStyle = function(s, trace) {
     // allow array marker and marker line colors to be
     // scaled by given max and min to colorscales
     var marker = trace.marker;
-    var markerScale = drawing.tryColorscale(marker, ''),
-        lineScale = drawing.tryColorscale(marker, 'line');
+    var markerScale = drawing.tryColorscale(marker, '');
+    var lineScale = drawing.tryColorscale(marker, 'line');
+    var gd = Lib.getPlotDiv(s.node());
```

Comment:
I wonder if it might be better to add a ref to those gradient defs in `fullData[i]` then to look for `gd`? Or even better refactor those `Drawing` method to expect `fullLayout` as argument.